pkg/controller/build/build_controller.go

```diff
@@ -996,6 +996,21 @@ func (ctrl *Controller) prepareForBuild(inputs *buildInputs) (ImageBuildRequest,
 // build, and updates the MachineConfigPool with an object reference for the
 // build pod.
 func (ctrl *Controller) startBuildForMachineConfigPool(ps *poolState) error {
+
+	if ctrlcommon.DoARebuild(ps.pool) {
+		// delete FAILED build attempts and builds
+		// delete rendered containerfile, MC, configmaps etc.
+		// Delete the actual build object itself.
+		// if we are rebuilding, we cannot ignore DNE. All of these objects should exist.
+		ctrl.postBuildCleanup(ps.pool, false)
```

Comment:
suggestion: If we want to not ignore DNE errors, check for and return any errors returned by `ctrl.postBuildCleanup()` here.